sources/model/entity/Calendar.ts

```diff
@@ -0,0 +1,94 @@
+import { Entity, PrimaryGeneratedColumn, ManyToOne, Column, ManyToMany, OneToMany, Repository } from "typeorm";
+import { Event } from "./Event"
+import { User } from "./User";
+
+@Entity("calendars")
+export class Calendar {
+
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @Column({length: 256})
+    name: string
+
+    @OneToMany(type => Event, event => event.calendar)
+    events: Event[]
+
+    @ManyToMany(type => User, user => user.calendars)
+    owners: User[]
+
+    @Column({name: "created_at", type: "timestamp with time zone", default: () => "CURRENT_TIMESTAMP"})
+    createdAt: Date;
+
+    @Column({name: "updated_at", type: "timestamp with time zone", default: () => "CURRENT_TIMESTAMP"})
+    updatedAt: Date;
+
+    public constructor(name: string, events: Event[], owners: User[]) {
+        this.name = name;
+        this.events = events;
+        this.owners = owners;
+    }
+
+    //
+    // Check if Calendar is valid
+    //
+    public isValid() : boolean {
+        return this.name.length > 0
+            && this.owners.length > 0;
+    }
+
+     /**
+     * @brief Create Calendar
+     *
+     * @param calendarRepository
+     * @param calendar
+     * 
+     * @returns The expected Calendar information on success
+     * @returns null on error
+     */
+    static async createCalendar(
+        calendarRepository: Repository<Calendar>,
+        calendar: Calendar
+    ) : Promise<Calendar> {
+        return calendarRepository.save(calendar);
+    }
+
+     /**
```

Comment:
Probably an indent error here (also in others comment blocks, check them)

It should be 4 spaces